rpctest/utils.go

```diff
@@ -76,24 +77,25 @@ func syncMempools(nodes []*Harness) error {
 	return nil
 }
 
-// syncBlocks blocks until all nodes report the same block height.
+// syncBlocks blocks until all nodes report the same best chain.
 func syncBlocks(nodes []*Harness) error {
 	blocksMatch := false
 
+retry:
 	for !blocksMatch {
-	retry:
-		blockHeights := make(map[int64]struct{})
+		blocksByHash := make(map[chainhash.Hash]int32)
 
 		for _, node := range nodes {
-			blockHeight, err := node.Node.GetBlockCount()
+			blockHash, blockHeight, err := node.Node.GetBestBlock()
 			if err != nil {
 				return err
 			}
 
-			blockHeights[blockHeight] = struct{}{}
-			if len(blockHeights) > 1 {
+			existingHeight, ok := blocksByHash[*blockHash]
```

Comment:
I think this code was a bit too clever (and hence hard to follow), so I changed it to something that I think should be much simpler.  It also has the benefit of being faster since it avoids the need for a map.